### PR TITLE
Bring docs/signatures in to agreement with cgo/BLAS

### PR DESCRIPTION
**Pull request by kortschak**
* Add license headers.
* Harmonise signatures.
* Harmonise previously missed panics.
* Make transpose behaviour conform to BLAS docs.

### CODE REVIEW

**Comment by kortschak** [comment]
@btracey PTAL

This should probably have been two. Sorry.

File: native/dgemm.go [hunk 1]
```diff
@@ -15,9 +15,16 @@ import (
 
 // Dgemm computes
 //  C = beta * C + alpha * A * B.
-// tA and tB specify whether A or B are transposed. A, B, and C are n×n dense
+// tA and tB specify whether A or B are transposed. A, B, and C are n×m dense
```
Comment (btracey) [inline native/dgemm.go]: The rest of the documentation uses m x n

File: native/dgemm.go [hunk 2]
```diff
@@ -15,9 +15,16 @@ import (
...
 func (Implementation) Dgemm(tA, tB blas.Transpose, m, n, k int, alpha float64, a []float64, lda int, b []float64, ldb int, beta float64, c []float64, ldc int) {
+	if tA != blas.NoTrans && tA != blas.Trans && tA != blas.ConjTrans {
+		panic(badTranspose)
+	}
+	if tB != blas.NoTrans && tB != blas.Trans && tB != blas.ConjTrans {
+		panic(badTranspose)
+	}
+
 	var amat, bmat, cmat general64
 	if tA == blas.Trans {
```
Comment (btracey) [inline native/dgemm.go]: This needs an extra Trans check (and below it)

**Comment by btracey** [comment]
A bunch of the level 2 functions are missing the "or tA == blas.ConjTrans"

**Comment by btracey** [comment]
Same with level 3

File: native/level2single.go [hunk 3]
```diff
@@ -1,5 +1,9 @@
 // Generated code do not edit. Run `go generate`.
 
+// Copyright ©2014 The gonum Authors. All rights reserved.
```
Comment (btracey) [inline native/level2single.go]: 2015, right?

File: native/level3double.go [hunk 4]
```diff
@@ -1,3 +1,7 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
```
Comment (btracey) [inline native/level3double.go]: 2015?

File: native/level3double_test.go [hunk 5]
```diff
@@ -1,3 +1,7 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
```
Comment (btracey) [inline native/level3double_test.go]: 2015

File: native/level3single.go [hunk 6]
```diff
@@ -1,5 +1,9 @@
 // Generated code do not edit. Run `go generate`.
 
+// Copyright ©2014 The gonum Authors. All rights reserved.
```
Comment (btracey) [inline native/level3single.go]: 2015

File: native/sgemm.go [hunk 7]
```diff
@@ -17,11 +17,18 @@ import (
 
 // Sgemm computes
 //  C = beta * C + alpha * A * B.
-// tA and tB specify whether A or B are transposed. A, B, and C are n×n dense
+// tA and tB specify whether A or B are transposed. A, B, and C are n×m dense
```
Comment (btracey) [inline native/sgemm.go]: m x n

**Comment by btracey** [comment]
LGTM after comments.

File: native/level2single.go (hunk 3 above)
Comment (kortschak) [inline native/level2single.go]: In all the cases where I have 2014, blame puts the earliest year as 2014.

Comment (btracey) [inline native/level2single.go]: Didn't you create level2single.go last week?

Comment (kortschak) [inline native/level2single.go]: A machine did, from the previously existing work. The translation code is 2015.

Comment (btracey) [inline native/level2single.go]: Okay, thanks.

**Pull request merged**